### ISSUE TITLE
Custom Mods in MSGFPlusAdapter

### ISSUE THREAD

**Issue by Stortebecker** [galaxyproteomics/tools-galaxyp#209]
Hi.

I did not find a possibility to add custom mods to the MSGFPlusAdapter. In OpenMS, users are expected to manually modify the mods.txt. However, I cannot find the file to modify in Galaxy. Our own MSGF+-Wrapper features a nice way to add custom mods, which I now also requested for MSGFPlusAdapter (see this [OpenMS Issue](https://github.com/OpenMS/OpenMS/issues/2977)).

It is not really high priority, as our own wrapper works fine, but it pretty much cripples the MSGFPlusAdapter for use-cases with stable isotope labelling. The latter are not listed in unimod.

**Comment by bgruening** [comment]
Does peptideshaker offers this?

**Comment by Stortebecker** [comment]
nope.

**Comment by glormph** [comment]
Custom mods work (I think) in the standalone MSGF+. Can one import mzIdentMLs to openMS from there?

**Comment by Stortebecker** [comment]
> It is not really high priority, as our own wrapper works fine, but it pretty much cripples the MSGFPlusAdapter for use-cases with stable isotope labelling. The latter are not listed in unimod.

**Correction:** Stable isotope labels **are** listed in unimod, but only for one aa each. Example: `Label 13C(6) (K)` and `Label 13C(6) (R)`, but not `Label 13C(6) (KR)` (a standard Silac label). I tested both by running MSGF+ directly and manually adding the corresponding entries to `Mods.txt`. There seems to be only minor differences in MSGFPlus runtime and output.

**But:** This does not solve our problem, as our wrapper for MSGFPlusAdapter only allows for one variable mod. I fixed this by deleting this restriction: https://github.com/galaxyproteomics/tools-galaxyp/pull/210

**Conclusion:** There is still no way to add custom mods in MSGFPlusAdapter, but the tool works for common labelling strategies, now. However, we can use our own MSGF+ wrapper for custom mods.
